runtimes/control.py

```
@@ -133,6 +133,15 @@ async def _start_local_runtime(runtime_id: str) -> dict[str, Any]:
             "error": "agent_runtime.py wrapper script not found",
         }
 
+    # Attempt to kill any existing process on the target port before starting
+    try:
+        if sys.platform == "win32":
+            subprocess.run(["powershell", "-Command", f"Stop-Process -Id (Get-NetTCPConnection -LocalPort {port}).OwningProcess -Force"], capture_output=True, check=False)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)